components/usePublishDate.js

```diff
@@ -0,0 +1,9 @@
+import { DateTime } from 'luxon';
+
+export const usePublishDate = (page) => {
+  const day = DateTime.fromISO(page.publish_date).setLocale('en-US').day;
+  const month = DateTime.fromISO(page.publish_date).setLocale('en-US').monthShort;
+  const year = DateTime.fromISO(page.publish_date).setLocale('en-US').year;
+  const publishDate = `Published on ${day}th of ${month}, ${year}`;
```

Comment:
Instead of formatting the date please refer to documentation of luxon [formatting](https://moment.github.io/luxon/docs/manual/formatting.html#the-basics)